sli/triedatum.cc

```diff
@@ -36,5 +36,5 @@ TrieDatum::equals( Datum const* dat ) const
 
   const TrieDatum* fd = dynamic_cast< TrieDatum* >( const_cast< Datum* >( dat ) );
 
-  return ( fd == NULL ) ? false : ( tree == fd->tree );
+  return fd != nullptr and tree == fd->tree;
```

Comment:
Not sure this is clearer now. Why not just have an `if/else` with two explicit `return` statements

Reply:
In my opinion, this line should stay as is. The first part of the statement simply ensures `fd` is defined, so that the following check does not try to dereference a `nullptr`.  
It could of course be expanded to 
```cpp
if (fd != nullptr)
{
    return tree == fd->tree;
}
return false;
```
which would need 5 more lines while making is barely more explicit? Or did you have something else in mind?

Reply:
I fear @jougs and I have different views on readability. Maybe I have written too much Python by now, but my preferred way of expressing this would be
```c++
return fd and fd->tree == tree;
```
Bjarne S himself writes "I prefer `if (p)` over `if (p != nullptr)` (§6.2.2, C++ Language, 4ed), so Python-style implicit conversion too bool is approved. And once one becomes used to this idiom (we should do it throughout in NEST ...), it is much more compact and with fewer degrees of freedom than

```c++
if ( fd )
{
  return tree == fd->tree;
}
else
{ 
  return false;
}
```
In the latter, I prefer an explicit `else` clause instead of falling of the end.

Reply:
Whatever. I can read both and it's deep in the SLI codebase. Let's just leave it as it is and get this merged.

Reply:
True, I also prefer leaving out the `!= nullptr`, much cleaner in my opinion. If you want, I can address those as well in the PR-sequel for the entire codebase.

Reply:
Fine by me.